arch/x86/core/fatal.c

```diff
@@ -142,19 +145,27 @@ static void unwind_stack(uintptr_t base_ptr, uint16_t cs)
 }
 #endif /* CONFIG_X86_EXCEPTION_STACK_TRACE */
 
-static inline struct x86_page_tables *get_ptables(const z_arch_esf_t *esf)
+static inline uintptr_t get_cr3(const z_arch_esf_t *esf)
 {
 #if defined(CONFIG_USERSPACE) && defined(CONFIG_X86_KPTI)
 	/* If the interrupted thread was in user mode, we did a page table
 	 * switch when we took the exception via z_x86_trampoline_to_kernel
 	 */
 	if ((esf->cs & 0x3) != 0) {
-		return z_x86_thread_page_tables_get(_current);
+		return _current->arch.ptables;
```

Comment:
dumb question, why cr3 is not pointing to this address in this situation ?

Reply:
if KPTI is turned on, we did a switch to the kernel's page table when we took the exception. we know that the thread page tables were active when the exception was generated so we return that.